File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    
    
    vector<int> answer;
    
 queue<int>q1 ;
    for(int i = 0 ; i < progresses.size();i++){
        
        if((100-progresses[i])%speeds[i]==0){
                q1.push((100 - progresses[i] ) / speeds[i]);
        }else{
                q1.push(((100 - progresses[i] ) / speeds[i])+1   );
        }
    
    }
    
    int count = 1;
    int a = q1.front();
    //cout << q1.front()<< "\n";
    q1.pop();
    for(int i=0;i<progresses.size()-1;i++){
        //cout << q1.front()<<  "\n";
        if(q1.front() <= a ){
            //cout << q1.front()<<"\n";
            q1.pop();
            count++;
            
        }else{
            answer.push_back(count);
            a = q1.front();
            q1.pop();
            count=1;
        }   
    }    
    answer.push_back(count);
    
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        //7 3 9
        // 5 10 1 1 20 1
        
        
        Stack<Integer> stack = new Stack<>();
        
        
        for(int i = progresses.length-1 ; i >= 0; i--){
            
            int remainProgress =  100 - progresses[i];
            int day = remainProgress / speeds[i];
            
            if(remainProgress % speeds[i] != 0){
                day+=1;
            }
            
            stack.add(day);
            
        }
    
        System.out.println(stack.peek());
        int before = stack.pop();
        
        List<Integer> answer = new ArrayList<>();
        
        int count = 1;
        

        while(!stack.empty()){
            
            System.out.println(stack.peek());
            if(stack.peek() <= before){
                count++;
                stack.pop();
            } else{
                answer.add(count);
                count = 1 ;
                before = stack.pop();
                
            }
            
            
            if(stack.empty()){
                answer.add(count);
            }
            
            
        }
        

    
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}